staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go

```diff
@@ -163,18 +212,35 @@ type RequestDigest struct {
 	User        user.Info
 }
 
-func (reqMgmt *requestManagementSystem) Wait(requestDigest RequestDigest) (execute bool, afterExecute func()) {
+func (reqMgmt *requestManagementSystem) Wait(requestDigest RequestDigest) (bool, func()) {
 	for {
-		rmState := reqMgmt.curState.Load().(*RequestManagementState)
+		rmState := reqMgmt.curState.Load().(*requestManagementState)
+
+		var matchingflowSchemaName string
+		var matchingpriorityLevelName string
+		var matchingFlowDistinguisherMethod *rmtypesv1alpha1.FlowDistinguisherMethod
+
+		// 1. computing flow
 		fs := rmState.pickFlowSchema(requestDigest)
-		// RequestManagementState is constructed to guarantee that the following succeeds
-		ps := rmState.priorityLevelStates[fs.Spec.PriorityLevelConfiguration.Name]
+		switch {
+		case fs != nil: // successfully matched a flow-schema
+			matchingflowSchemaName = fs.Name
+			matchingpriorityLevelName = fs.Spec.PriorityLevelConfiguration.Name
+			matchingFlowDistinguisherMethod = fs.Spec.DistinguisherMethod
+		default: // reject
+			return false, func() {}
+		}
+
+		// 2. computing hash
+		flowDistinguisher := requestDigest.ComputeFlowDistinguisher(matchingFlowDistinguisherMethod)
+		hashValue := hashFlowID(matchingflowSchemaName, flowDistinguisher)
+
+		// 3. executing
+		ps := rmState.priorityLevelStates[matchingpriorityLevelName]
 		if ps.config.Exempt {
 			klog.V(7).Infof("Serving %v without delay", requestDigest)
```

Comment:
This produces log entries like the following.
```
I0815 21:01:28.555153    5092 reqmgmt.go:241] Serving {0xc0082711e0 0xc003cc2640} without delay
```
--- which are pretty uninformative.  The more verbose log statements like
```
			klog.V(6).Infof("Serving RequestInfo=%#+v, user.Info=%#+v after queuing\n", requestInfo, user)
```
produce more informative log statements, like
```
I0815 21:01:28.555169    5092 reqmgmt.go:70] Serving RequestInfo=&request.RequestInfo{IsResourceRequest:true, Path:"/apis/flowcontrol.apiserver.k8s.io/v1alpha1/prioritylevelconfigurations", Verb:"list", APIPrefix:"apis", APIGroup:"flowcontrol.apiserver.k8s.io", APIVersion:"v1alpha1", Namespace:"", Resource:"prioritylevelconfigurations", Subresource:"", Name:"", Parts:[]string{"prioritylevelconfigurations"}}, user.Info=&user.DefaultInfo{Name:"system:apiserver", UID:"0c86221d-db07-40dd-b150-9fa17227bbef", Groups:[]string{"system:masters"}, Extra:map[string][]string(nil)} after queuing
```